### PR TITLE
Add security profile tests and fixtures

### PR DESCRIPTION
**Pull request by armintaenzertng**
also fixed some minor bugs in the class definitions.

part of #432 and #426

### CODE REVIEW

**Comment by HarshvMahawar** [comment]
@armintaenzertng can you please explain what are these vulnerability assessment relationships?

**Comment by armintaenzertng** [comment]
Have a look in the official markdown files: https://github.com/spdx/spdx-3-model/tree/main/model/Security/Classes
The Security profile mainly consists of these assessment relationships. Each class comes with its own description that explains what it is.

File: src/spdx_tools/spdx3/model/security/cvss_v2_vuln_assessment_relationship.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 @dataclass_with_properties
 class CvssV2VulnAssessmentRelationship(VulnAssessmentRelationship):
-    score: float = None
+    score: str = None
```
Comment (meretp) [inline src/spdx_tools/spdx3/model/security/cvss_v2_vuln_assessment_relationship.py:22]: Needs to be clarified as the markdown files in the spdx-3-model state `str` as well as `xsd:decimal`.
-> https://github.com/spdx/spdx-3-model/issues/332

**Review by meretp — APPROVED** [review]
Thanks, this looks really nice with the refactoring you did before. I think we can merge this and update once the issue in the spdx-3-model is solved.

**Pull request merged**